ExampleCheckout/UITests/ExtraElementsTests.swift

```diff
@@ -0,0 +1,69 @@
+
+final class ExtraElementsTests: NetworksTests {
+    func testExtraElements_whenNone_shouldNotDisplay() throws {
+        try XCTContext.runActivity(named: "Open payment form") { _ in
+            try startPaymentSession(transaction: Transaction())
+            app.staticTexts["Cards"].tap()
+            XCTAssertFalse(app.textViews["Top Element Number 1 with invalid link"].exists)
+            XCTAssertFalse(app.textViews["Bottom Element Number 2 without Checkbox"].exists)
+        }
+    }
+
+    func testExtraElements_whenTopAndBottom_shouldDisplayBoth() throws {
+        try XCTContext.runActivity(named: "Open payment form") { _ in
+            try startPaymentSession(transaction: Transaction(), url: topBottomElementsURL)
+            app.tables.staticTexts["Cards"].tap()
+            XCTAssertTrue(app.textViews["Top Element Number 1 with invalid link"].exists)
+            XCTAssertTrue(app.textViews["Bottom Element Number 2 without Checkbox"].exists)
+        }
+    }
+
+    func testExtraElements_whenTop_shouldDisplayTop() throws {
+        try XCTContext.runActivity(named: "Open payment form") { _ in
+            try startPaymentSession(transaction: Transaction(), url: topElementURL)
+            app.tables.staticTexts["Cards"].tap()
+            XCTAssertTrue(app.textViews["Top Element Number 1 with invalid link"].exists)
+            XCTAssertFalse(app.textViews["Bottom Element Number 2 without Checkbox"].exists)
+        }
+    }
+
+    func testExtraElements_whenBottom_shouldDisplayBottom() throws {
+        try XCTContext.runActivity(named: "Open payment form") { _ in
+            try startPaymentSession(transaction: Transaction(), url: bottomElementURL)
+            app.tables.staticTexts["Cards"].tap()
+            XCTAssertTrue(app.textViews["Bottom Element Number 2 without Checkbox"].exists)
+            XCTAssertFalse(app.textViews["Top Element Number 1 with invalid link"].exists)
+        }
+    }
+
+    func testExtraElements_whenContainsCheckbox_shouldNotDisplay() throws {
+        try XCTContext.runActivity(named: "Open payment form") { _ in
+            try startPaymentSession(transaction: Transaction(), url: topBottomElementsURL)
+            app.tables.staticTexts["Cards"].tap()
+            XCTAssertFalse(app.textViews["Top Element Number 2 with a Checkbox view"].exists)
+            XCTAssertFalse(app.textViews["Bottom Element Number 1 with an Checkbox optional title"].exists)
+        }
+    }
+
+    func testExtraElements_whenContainsLink_shouldOpenSafari() throws {
+        try XCTContext.runActivity(named: "Open payment form") { _ in
+            try startPaymentSession(transaction: Transaction(), url: topBottomElementsURL)
+            app.staticTexts["Cards"].tap()
+            app.textViews.firstMatch.links["Number 1"].tap()
+            XCTAssertTrue(app.webViews.firstMatch.waitForExistence(timeout: 1))
```

Comment:
The thing I want to suggest is to increase the timeout because CI environment could be very laggy even in view controller loading time.

```suggestion
            XCTAssertTrue(app.webViews.firstMatch.waitForExistence(timeout: 5))
```

Reply:
@siebeprojects taught me not to use any non-named numbers in the code (sometimes I still fail with that). I don't know if it is applicable here but it could make the number more descriptive.

```suggestion
            let safariViewControllerPresentationTimeout = 5
            XCTAssertTrue(app.webViews.firstMatch.waitForExistence(timeout: safariViewControllerPresentationTimeout))
```

_Not a blocker (from my side)._

Reply:
> @siebeprojects taught me not to use any non-named numbers in the code

100% agree, magic numbers are bad. In test code I don't usually worry too much about this sort of thing, but we could have a "global" timeout constant for tests.

5 is A LOT though. If we have 100 tests each waiting 5 seconds before failing, it can add a lot of extra time to each test run. Whenever I've had to use more than 1 second timeout in a test, it was usually because something was wrong somewhere in the implementation. I can extract this but I'd keep it to 1 second and if a test fails because of that we can investigate or adjust it.

Reply:
5 seconds is not a lot for our CI. Sometimes just animating the presentation of the new view controller could take up to 10 seconds :), I caught timeout errors multiple times. You could extract it to some global property if you want, we just don't use it anywhere else, so I don't think it makes sense yet.

Reply:
Addressed in https://github.com/optile/checkout-ios/pull/137/commits/ab1b24d09cbaf1048ba08ffba4ee2f2be2bb2187.